utils/aggregateData.ts

```
@@ -0,0 +1,120 @@
+        if (!!+grCat && +grCat !== +grId) return;
+
+        tR[dateKey][`${grId}-0`] ??= [0];
+        tR[dateKey][`${grId}-0`][0] += valueArr[0];
+      }
+    });
+  });
+
+  return tR;
+};
+
+const aggregateData = (expenses: [], categories: {}) => {
```

Biome

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)